odoo/addons/base/models/ir_model.py

```diff
@@ -1549,15 +1547,7 @@ def _update_xmlids(self, data_list, update=False):
 
         for sub_rows in self.env.cr.split_for_in_conditions(rows):
             # insert rows or update them
-            query = """
-                INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)
-                VALUES {rows}
-                ON CONFLICT (module, name)
-                DO UPDATE SET date_update=(now() at time zone 'UTC') {where}
-            """.format(
-                rows=", ".join([rowf] * len(sub_rows)),
-                where="WHERE NOT ir_model_data.noupdate" if update else "",
-            )
+            query = self._build_update_xmlids_query(len(sub_rows), update)
```

Comment:
I'd pass directly `sub_rows` as a parameter for flexibility in case it is needed in the future.

Reply:
Done